source/Crew.cpp

```diff
@@ -0,0 +1,232 @@
+/* Crew.cpp
+Copyright (c) 2019 by Luke Arndt
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Crew.h"
+#include "Files.h" 
+#include "GameData.h"
+
+using namespace std;
+
+void Crew::Load(const DataNode &node)
+{
+	if(node.Size() >= 2)
+		id = node.Token(1);
+	
+	for(const DataNode &child : node)
+	{
+		if(child.Size() >= 2)
+		{
+			if(child.Token(0) == "name")
+				name = child.Token(1);
+			else if(child.Token(0) == "minimum per ship")
+				minimumPerShip = max((int)child.Value(1), 0);
+			else if(child.Token(0) == "parked salary")
+				parkedSalary = max((int)child.Value(1), 0);
+			else if(child.Token(0) == "population per member")
+				populationPerMember = max((int)child.Value(1), 0);
+			else if(child.Token(0) == "salary")
+				salary = max((int)child.Value(1), 0);
+			else
+				child.PrintTrace("Skipping unrecognized attribute:");
+		}
+		else if(child.Token(0) == "avoids escorts")
+			avoidsEscorts = true;
+		else if(child.Token(0) == "avoids flagship")
+			avoidsFlagship = true;
+		else
+			child.PrintTrace("Skipping incomplete attribute:");
+	}
+}
+
+
+
+int64_t Crew::CalculateSalaries(const vector<shared_ptr<Ship>> &ships, const Ship * flagship, const bool includeExtras)
+{
+	bool isFlagship = false;
+	bool checkIfFlagship = true;
+	int64_t totalSalaries = 0;
+	
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		if(checkIfFlagship)
```

Comment:
For maintainability of the code I would suggest to define the boolean isFlagship here within the loop and remove the "checkIfFlagship" variable completely.

Reply:
This was a micro performance optimisation; it does add a fair amount of clutter though. Probably isn't worth it.